context-free.md

```diff
@@ -1,22 +1,42 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+ContextFree is for people who are interested in making (variations of ) cool 
+pictures, with minimal effort.
 
 ## What is easy to do in this language? Why is it easy?
-
+It's easy to create complex images from deceptively simple rules, thanks to 
+repeated patterns, recursion, and built-in randomness.
 
 ## What is hard to do in this language? Why is it hard?
-
+It's hard to have complete control over the layout of a picture.  If you wanted
+to write words or draw an image that requires pixel-level precision, you'd have
+a hard time. The reason it's hard is because randomness is a key aspect of
+ContextFree: the idea is to be able to make many different variations on a theme.
+The language encourages play over rigor.
```

Comment:
I also ran into these issues in my version of the assignment - it seems like you came up with a good design that plays towards the language's strengths, and avoids its weaknesses